File: Algorithms/Searching-Algorithms/binary-search/binarysearch.java
import java.util.Arrays;
public class binarysearch {

    public static int binarySearch(int[] arr, int target){
        int low = 0;
        int high = arr.length - 1;

        while(low <= high){
            int mid = (low + high)/ 2;

            if(arr[mid] == target){
                return mid;
            }else if(arr[mid] < target){
                low = mid + 1;
            }
                
            else{
                high = mid -1;
            }
        }
        return -1;
    }
    
    public static void main(String[] args){
        int[] arr = {1,260,23,54,7,8,89,0};
        Arrays.sort(arr);
        int target = 54;
        

        int result = binarySearch(arr, target);
        if(result == -1){
            System.out.println("Element not found");
        }
        else{
            System.out.println("Element " + target + " found at the index " + result);
        }
    }


}

File: Algorithms/Searching-Algorithms/binary-search/binarysearch.py
def binary_search(arr, target):
  
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1


def main():
    n = int(input("Enter the number of elements: "))

    arr = []
    print(f"Enter {n} elements:")
    for _ in range(n):
        arr.append(int(input()))

    # Binary search requires a sorted array
    arr.sort()
    print("Sorted array:", arr)

    target = int(input("Enter the target element: "))

    result = binary_search(arr, target)

    if result == -1:
        print("Element not found in the array.")
    else:
        print(f"Element {target} found at index {result}.")


if __name__ == "__main__":
    main()
